### DISCUSSION TITLE
Update scrollarea thumb size programmatically

### DISCUSSION THREAD

**Discussion by fdb75017** [mantinedev/mantine#5038] [Q&A] [answered]
Hello @rtivital ,

First of all, thanks a lot for leading this amazing lib ! I use and appreciate it a lot ❤️ !!!

I'm currently trying to use scrollarea within a "react-virtuoso" grid, together with infinite loading. I'm seeing an issue which I think I understand: when loading more items, because of virtualization, the actual content of the scrollarea does not necessarily change, which prevents the scrollarea component from recomputing the size of the thumb and gives a buggy impression. For example we can still scroll the area while the thumb has stayed at the bottom of the scrollbar.

I guess this is a case where we would need to be able to tell the scrollarea to recompute the thumbs height/width programatically. Looking at the various apis from the component I do not yet see a way around this. I'm using type="hover" so I've tried a hacky pointerleave => timeout => pointerenter pipe but this really is ugly (and does not work eg if you leave the area before the thumb has disappeared...)

Do you see another way to do this ?
Many thanks in advance !

**Comment by rtivital** [comment]
To force rerender a component you can set `key` prop

**Reply by fdb75017** [reply]
Ah yes good point, thanks for your answer !
However enforcing the complete re-render of the `ScrollArea` does scroll it back to its initial position (`scrollTop=0`) which is not what you'd expect for an infinite loading component. I've tried to log the `scrollTop` just before changing the key, then timeout before re-scrolling "manually" using the `viewportRef` but it sounds a bit hacky and does not work if you're infinite loading from the top (like a chat component). I guess i'd need to be able to tell the `ScrollArea` to just update the scroll thumb position/size....

**Reply by rtivital** [reply, marked as answer]
I do not really know what to do in this case then, it is probably better to use native scroll container instead for infinite scrolling
